localstack/services/stepfunctions/asl/eval/count_down_latch.py

```diff
@@ -0,0 +1,21 @@
+import threading
```

Comment:
minor: Wondering if we could create a cleaner separation between the ASL language artifacts (incl. grammar, lexer, parser, and generated code), and the execution runtime. Currently, the concepts are a bit interwoven in the code structure. 

It seems like a first quick win could be to pull out the `stepfunctions.asl.eval` module out into a `stepfunctions.runtime` module - which would more cleanly separate the concepts.

Reply:
Excellent point. In ext we have an iteration of this project which separated components into `Component`s and `Evaluation`. This seems similar to the idea you mention here. 

`Evaluation` component used to decorate `Components`, thereby separating the two aspects. As the `Component`s abstraction appeared to duplicate `ANTLR`'s 'nodes' abstraction, and given the added amount of files it involved (2 per component) and verbose code (constructors mainly), the decision was made to have `EvalComponents` extend `Components`. Another minor aspect to address in that setup was how some Components do not require an evaluation logic, but are members of other `Component`s.

I still second the idea of having this separation, but I see how this would require a more careful design first.